python/lsst/ts/aos/analysis/nightlyReport.py

```diff
@@ -0,0 +1,591 @@
+from tqdm import tqdm as tqdm
+import galsim
+from lsst.summit.utils import (
+    ConsDbClient,
+    getBandpassSeeingCorrection,
+    getAirmassSeeingCorrection,
+    makeDefaultButler,
+)
+from lsst.ts.ofc import (
+    OFC,
+    OFCData,
+    OICController,
+    StateEstimator,
+    SensitivityMatrix,
+    BendModeToForce,
+)
+from lsst.summit.utils.efdUtils import (
+    makeEfdClient,
+    getEfdData,
+    getMostRecentRowWithDataBefore,
+)
+from astropy.time import Time, TimeDelta
+from lsst.summit.extras.ringssSeeing import RingssSeeingMonitor
+from lsst.ts.wep.utils import convertZernikesToPsfWidth, makeDense
+from lsst.obs.lsst import LsstCam
+import warnings
+
+warnings.filterwarnings("ignore")
+__all__ = [
+    "compute_radial_profile_ratio",
+    "compute_radial_profile",
+    "compute_spectral_metrics",
+    "get_band_mapping",
+    "AOSDatabase",
+    "plot_statistics",
+]
+
+
+class AOSDatabase:
+    def __init__(
+        self,
+        seq_max: int = 9999,
+        consdb_url: str = "http://consdb-pq.consdb:8080/consdb",
+    ) -> None:
+        """Create fetcher.
+
+        Parameters
+        ----------
+        seq_max : int, optional
+            Maximum sequence number to fetch. Default is 9999.
+        consdb_url : str, optional
+            URL to create ConsDB client.
+            The default is "http://consdb-pq.consdb:8080/consdb".
+        """
+        butler = Butler(
+            "LSSTCam",
+            collections=["LSSTCam/runs/quickLook", "LSSTCam/raw/all"],
+            instrument="LSSTCam",
+        )
+        self.butler = butler
+
+        # Create the EFD, ConsDB, Seeing clients
+        self.efd_client = makeEfdClient()
+        self.cdb_client = ConsDbClient(consdb_url)
+        self.seeing_monitor = RingssSeeingMonitor(self.efd_client)
+
+        self.ofc_data = OFCData("lsst")
+        self.ofc_data.controller["truncation_index"] = 6
+
+        m2_hexapod = np.ones(5, dtype=bool)
+        cam_hexapod = np.ones(5, dtype=bool)
+        m1m3_bending = np.zeros(20, dtype=bool)
+        m2_bending = np.zeros(20, dtype=bool)
+        m1m3_bending[:5] = True
+        m2_bending[:5] = True
+        self.ofc_data.comp_dof_idx = dict(
+            m2HexPos=m2_hexapod,
+            camHexPos=cam_hexapod,
+            M1M3Bend=m1m3_bending,
+            M2Bend=m2_bending,
+        )
+        self.state_estimator = StateEstimator(self.ofc_data)
+
+        self.m2_bmf = BendModeToForce("M2", self.ofc_data)
+        self.m1m3_bmf = BendModeToForce("M1M3", self.ofc_data)
+
+        self.det_order = [191, 195, 199, 203]
+        camera = LsstCam().getCamera()
+        self.detector_names = [
+            camera.get(det_id).getName() for det_id in self.det_order
+        ]
+
+        dz_sensitivity_matrix = SensitivityMatrix(self.ofc_data)
+
+        # Create initial database
+        self.data = None
+        self.seq_max = seq_max
+
+    @classmethod
+    async def create(
+        cls,
+        day,
```

Comment:
I think renaming this everywhere to `day_obs` would make it a lot more consistent with other observatory code.